### PR TITLE
Replace deprecated Gtk.TreeView with ListView: Activity Parameter Nodes

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [ ] Feature
- [x] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

`Gtk.TreeView` is deprecated. GTK 4 has a much nicer tree widget: `Gtk.ListVIew`. It's already used for the model browser. We should also use it in the property editor, for both consistency and ease of use.

### Does this PR introduce a breaking change?
- [x] Yes
- [ ] No

### Other information

`Gtk.ListView` has no header. This will be [introduced in GTK 4.12](https://docs.gtk.org/gtk4/property.ListView.header-factory.html). So we may want to postpone changing to `ListView`s.

### CODE REVIEW

**Comment by amolenaar** [comment]
This PR now has replaced the activity parameter property page with a new style list view.

I think it reached feature parity, with keybindings and easy-adding features in place.

Let's do the other TreeView's via different PR's, so we can keep the change manageable.

File: gaphor/UML/actions/tests/test_activitypropertypage.py [hunk 1]
```diff
@@ -1,53 +1,185 @@
...
+    property_page.construct()
+
+    del activity_item.subject.node[node]
+
+    assert not activity_item.subject.node
+    assert property_page.model.get_n_items() == 1
+    assert property_page.model.get_item(0).node is None
+
+
+def test_update_model_with_multiple_noes_when_parameter_deleted(
```
Comment (danyeaw) [inline gaphor/UML/actions/tests/test_activitypropertypage.py:107]: Super nit, but should this be nodes instead of noes?

**Review by danyeaw — APPROVED** [review]
Really nice update to use ListView @amolenaar!

File: gaphor/UML/actions/tests/test_activitypropertypage.py (hunk 1 above)
Comment (amolenaar) [reply gaphor/UML/actions/tests/test_activitypropertypage.py]: fixed

**Pull request merged**